File: docs/iOS/YUndoManager.swift
import Foundation
import JavaScriptCore

public class YUndoManager {
  
  let jsValue: JSValue
  
  init(jsValue: JSValue) {
    self.jsValue = jsValue
  }
  
  public func undo() {
    jsValue.invokeMethod("undo", withArguments: [])
  }
  
  public func redo() {
    jsValue.invokeMethod("redo", withArguments: [])
  }
}

File: library/src/main/java/io/agora/board/yjs/YUndoManager.kt
package io.agora.board.yjs

import com.hippo.quickjs.android.JSContext
import com.hippo.quickjs.android.JSValue

/**
 * author : fenglibin
 * date : 2024/7/20
 * description :
 */
class YUndoManager(private val jsValue: JSValue, private val context: JSContext) {
    fun undo() {
        val f = jsValue.getKeyedFunction("undo")
        context.callFunction(f, jsValue, arrayOf())
    }

    fun redo() {
        val f = jsValue.getKeyedFunction("redo")
        context.callFunction(f, jsValue, arrayOf())
    }
}